static/js/webrtc.js

```diff
@@ -234,22 +240,40 @@ var rtc = (function() {
           videoContainer.addClass('local-user');
           video.muted = true;
         }
-        self.addInterface(userId);
+        self.addInterface(userId, stream);
       }
       if (stream) {
         attachMediaStream(video, stream);
       } else if (video) {
         $(video).parent().remove();
       }
     },
-    addInterface: function(userId) {
+    addInterface: function(userId, stream) {
       var isLocal = userId == self.getUserId();
       var videoId = "video_" + userId.replace(/\./g, "_");
       var $video = $("#" + videoId);
 
+      ///////
+      // Mute button
+      ///////
+
+      var audioTrack = stream.getAudioTracks()[0];
+      var initiallyMuted = true; // if there's no audio track, it's muted
+      if (audioTrack) {
+        initiallyMuted = !audioTrack.enabled
+      }
+
       var $mute = $("<span class='interface-btn audio-btn buttonicon'>")
-        .attr("title", "Mute")
-        .on({
+        .attr("title", clientVars.webrtc.audio_allowed
+          ? (initiallyMuted ? "Unmute" : "Mute")
+          : "Audio disallowed by admin")
+        .toggleClass("muted", initiallyMuted)
+        .toggleClass("disallowed", !clientVars.webrtc.audio_allowed);
+
+      // TODO - is audio_enabled/video_enabled a security issue? Like do we care if the user hacks the client to do video anyway?
+      // I suppose this has nothing to do with the server anyway. It uses Google turn servers etc
```

Comment:
@JohnMcLear If I understand webrtc correctly, it doesn't touch the etherpad-lite server at all, right? Just these stun and turn servers which are usually run by somebody else?

I only ask to make sure we're not missing any security requirements server-side. But if I understand webrtc right, I think that there's nothing we can do to stop some tricky user from turning on audio and video chat despite the site admin's wishes.

Reply:
Correct.  and wait until you have to debug TURN/STUN that's a whole new train of pain :D

How would the tricky user do that btw?

Reply:
Oh I mean if the client is exploited that's way beyond the scope of this project.  We provide an interface to the browser.  If the browser as the client is exploited there is nothing we can do..

Reply:
Right. I mean they could likely figure something out via the debug console.

Reply:
Yeah I think the only thing that could go wrong is that two sneaky users of the etherpad installation would do a video chat against the wishes of the admin.

Now that I understand that the request to initiate the RTC connection is sent over the server (i.e. my comment is wrong), perhaps we could do something on the server side to block that initiation. But, I imagine it's not a high priority.